### PR TITLE
[Erlang] RollDice-related doc improvements

### PR DESCRIPTION
**Pull request by apreifsteck**
Light edits to the documentation to streamline the getting started guide, and adds some tips and gotchas.

More details:
- Fixes broken link pointing to the RollDice example project
- Changes code examples to use `RollDice` instead of `DiceGame` (DiceGame outdated)
- Explicitly adds in a few implied code changes in the getting started guide. It should be more copy-paste ready now.
- Adds an in-line docker compose file and collector config to the exporting guide, for easy copy-pasta

### CODE REVIEW

**Comment by theletterf** [comment]
@apreifsteck Hi there! Could you please sign the CLA and add a bit of context in the description? Thanks!

**Comment by chalin** [comment]
Thanks for this @apreifsteck. I'll let @open-telemetry/erlang-approvers comment about the content.

Note that a few of the GitHub checks are failing. You'll need to pull this PR branch back down, rebase from `main`, and fix the issues. For guidance, see https://opentelemetry.io/docs/contributing/#fix-content-issues-automatically.

File: content/en/docs/languages/erlang/instrumentation.md [hunk 1]
```diff
@@ -17,12 +17,18 @@ Add the following dependencies to your project:
 - `opentelemetry`: contains the SDK that implements the interfaces defined in
   the API. Without it, all the functions in the API are no-ops.
 
+Optionally, you may choose to include the `open_telemetry_decorator` library.
+Note that this library is not officially supported by the opentelemetry project,
```
Comment (tsloughter) [inline content/en/docs/languages/erlang/instrumentation.md:21]: This should be removed since not officially supported.

**Comment by tsloughter** [comment]
Changes look ok and are important (module names are wrong currently) but one thing should be removed that I noted.

**Review by svrnm — APPROVED** [review]

**Review by tsloughter — APPROVED** [review]

**Pull request merged**